luma/core/virtual.py

```diff
@@ -227,8 +229,14 @@ def println(self, text=""):
         Prints the supplied text to the device, scrolling where necessary.
         The text is always followed by a newline.
         """
-        self.puts(text)
-        self.newline()
+        if self.word_wrap:
+            text = wrap(text, width=self.width)
```

Comment:
I've added a test but I'm not sure if it's correct, probably not and we need to ignore out these ansi color definitions?

![ansi_colors_wrapped](https://cloud.githubusercontent.com/assets/305679/25072278/9be2f5c4-22ca-11e7-9371-20a91dc1d4fc.png)